src/compiler/parser.ts

```diff
@@ -2271,7 +2295,13 @@ namespace ts {
                 isStartOfType(/*inStartOfParameter*/ true);
         }
 
-        function parseParameter(requireEqualsToken?: boolean): ParameterDeclaration {
+        function parseParameterInSpeculation(): ParameterDeclaration | Fail {

```

Comment:
Not important, but I'd be tempted to call this one `tryParseParameter` and the other one `parseParameter`. We generally use `tryParse` to indicate speculation vs. `parse` to indicate definitive parsing, though that's not always the case.